plugins/inputs/nfsclient/nfsclient.go

```diff
@@ -0,0 +1,421 @@
+	"vfsreadpage",
+	"vfsreadpages",
+	"vfswritepage",
+	"vfswritepages",
+	"vfsgetdents",
+	"vfssetattr",
+	"vfsflush",
+	"vfsfsync",
+	"vfslock",
+	"vfsrelease",
+	"congestionwait",
+	"setattrtrunc",
+	"extendwrite",
+	"sillyrenames",
+	"shortreads",
+	"shortwrites",
+	"delay",
+	"pnfsreads",
+	"pnfswrites",
+}
+
+var bytesFields = []string{
+	"normalreadbytes",
+	"normalwritebytes",
+	"directreadbytes",
+	"directwritebytes",
+	"serverreadbytes",
+	"serverwritebytes",
+	"readpages",
+	"writepages",
+}
+
+var xprtudpFields = []string{
+	"bind_count",
+	"rpcsends",
+	"rpcreceives",
+	"badxids",
+	"inflightsends",
+	"backlogutil",
+}
+
+var xprttcpFields = []string{
+	"bind_count",
+	"connect_count",
+	"connect_time",
+	"idle_time",
+	"rpcsends",
+	"rpcreceives",
+	"badxids",
+	"inflightsends",
+	"backlogutil",
+}
+
+var nfs3Fields = []string{
+	"NULL",
+	"GETATTR",
+	"SETATTR",
+	"LOOKUP",
+	"ACCESS",
+	"READLINK",
+	"READ",
+	"WRITE",
+	"CREATE",
+	"MKDIR",
+	"SYMLINK",
+	"MKNOD",
+	"REMOVE",
+	"RMDIR",
+	"RENAME",
+	"LINK",
+	"READDIR",
+	"READDIRPLUS",
+	"FSSTAT",
+	"FSINFO",
+	"PATHCONF",
+	"COMMIT",
+}
+
+var nfs4Fields = []string{
+	"NULL",
+	"READ",
+	"WRITE",
+	"COMMIT",
+	"OPEN",
+	"OPEN_CONFIRM",
+	"OPEN_NOATTR",
+	"OPEN_DOWNGRADE",
+	"CLOSE",
+	"SETATTR",
+	"FSINFO",
+	"RENEW",
+	"SETCLIENTID",
+	"SETCLIENTID_CONFIRM",
+	"LOCK",
+	"LOCKT",
+	"LOCKU",
+	"ACCESS",
+	"GETATTR",
+	"LOOKUP",
+	"LOOKUP_ROOT",
+	"REMOVE",
+	"RENAME",
+	"LINK",
+	"SYMLINK",
+	"CREATE",
+	"PATHCONF",
+	"STATFS",
+	"READLINK",
+	"READDIR",
+	"SERVER_CAPS",
+	"DELEGRETURN",
+	"GETACL",
+	"SETACL",
+	"FS_LOCATIONS",
+	"RELEASE_LOCKOWNER",
+	"SECINFO",
+	"FSID_PRESENT",
+	"EXCHANGE_ID",
+	"CREATE_SESSION",
+	"DESTROY_SESSION",
+	"SEQUENCE",
+	"GET_LEASE_TIME",
+	"RECLAIM_COMPLETE",
+	"LAYOUTGET",
+	"GETDEVICEINFO",
+	"LAYOUTCOMMIT",
+	"LAYOUTRETURN",
+	"SECINFO_NO_NAME",
+	"TEST_STATEID",
+	"FREE_STATEID",
+	"GETDEVICELIST",
+	"BIND_CONN_TO_SESSION",
+	"DESTROY_CLIENTID",
+	"SEEK",
+	"ALLOCATE",
+	"DEALLOCATE",
+	"LAYOUTSTATS",
+	"CLONE",
+}
+
+var nfsopFields = []string{
+	"ops",
+	"trans",
+	"timeouts",
+	"bytes_sent",
+	"bytes_recv",
+	"queue_time",
+	"response_time",
+	"total_time",
+	"errors",
+}
+
+func convert(line []string) []int64 {
+	var nline []int64
+	for _, l := range line[1:] {
+		f, _ := strconv.ParseInt(l, 10, 64)
+		nline = append(nline, f)
+	}
+	return nline
+}
+
+func in(list []string, val string) bool {
+	for _, v := range list {
+		if v == val {
+			return true
+		}
+	}
+	return false
+}
+
+func (n *NFSClient) parseStat(mountpoint string, export string, version string, line []string, fullstat bool, nfs3Ops map[string]bool, nfs4Ops map[string]bool, acc telegraf.Accumulator) error {
+	tags := map[string]string{"mountpoint": mountpoint, "serverexport": export}
+	nline := convert(line)
+	first := strings.Replace(line[0], ":", "", 1)
+
+	var fields = make(map[string]interface{})
+
+	if fullstat && first == "events" && len(nline) >= len(eventsFields) {
```

Comment:
For this function, please move the non-fullstat stuff up here and return in the if. Then afterwards, you can skip the check for `fullstat`.

Reply:
Actually, this is already done, and I refactored slightly.  The "non-fullstat" metrics are always returned, and a simple "if fullstat {}" block encompasses the rest of the parsing.

Reply:
Refactored so that there is a single check for `fullstat`, after the mandatory collections.